homeassistant/components/tahoma/cover.py

```diff
@@ -0,0 +1,342 @@
+    "Window": DEVICE_CLASS_WINDOW,
+}
+
+
+async def async_setup_entry(hass, entry, async_add_entities):
+    """Set up the TaHoma covers from a config entry."""
+    data = hass.data[DOMAIN][entry.entry_id]
+    coordinator = data["coordinator"]
+
+    entities = [
+        TahomaCover(device.deviceurl, coordinator)
+        for device in data["platforms"][COVER]
+    ]
+
+    async_add_entities(entities)
+
+
+class TahomaCover(TahomaEntity, CoverEntity):
+    """Representation of a TaHoma Cover."""
+
+    @property
+    def current_cover_position(self):
+        """
+        Return current position of cover.
+
+        None is unknown, 0 is closed, 100 is fully open.
+        """
+        position = self.executor.select_state(
+            CORE_CLOSURE_STATE,
+            CORE_DEPLOYMENT_STATE,
+            CORE_PEDESTRIAN_POSITION_STATE,
+            CORE_TARGET_CLOSURE_STATE,
+            CORE_CLOSURE_OR_ROCKER_POSITION_STATE,
+        )
+
+        # Uno devices can have a position not in 0 to 100 range when unknown
+        if position is None or position < 0 or position > 100:
+            return None
+
+        if not self._reversed_position_device():
+            position = 100 - position
+
+        return position
+
+    @property
+    def current_cover_tilt_position(self):
+        """Return current position of cover tilt.
+
+        None is unknown, 0 is closed, 100 is fully open.
+        """
+        position = self.executor.select_state(
+            CORE_SLATS_ORIENTATION_STATE, CORE_SLATE_ORIENTATION_STATE
+        )
+        return 100 - position if position is not None else None
+
+    async def async_set_cover_position(self, **kwargs):
+        """Move the cover to a specific position."""
+        position = 100 - kwargs.get[ATTR_POSITION]
+
+        if self._reversed_position_device():
+            position = kwargs.get[ATTR_POSITION]
+
+        await self.executor.async_execute_command(
+            self.executor.select_command(*COMMANDS_SET_POSITION), position
+        )
+
+    async def async_set_cover_tilt_position(self, **kwargs):
+        """Move the cover tilt to a specific position."""
+        await self.executor.async_execute_command(
+            self.executor.select_command(*COMMANDS_SET_TILT_POSITION),
+            100 - kwargs[ATTR_TILT_POSITION],
+        )
+
+    @property
+    def is_closed(self):
+        """Return if the cover is closed."""
+        state = self.executor.select_state(
+            CORE_OPEN_CLOSED_STATE,
+            CORE_SLATS_OPEN_CLOSED_STATE,
+            CORE_OPEN_CLOSED_PARTIAL_STATE,
+            CORE_OPEN_CLOSED_PEDESTRIAN_STATE,
+            CORE_OPEN_CLOSED_UNKNOWN_STATE,
+            MYFOX_SHUTTER_STATUS_STATE,
+        )
+
+        if state is not None:
+            return state == STATE_CLOSED
+
+        if self.current_cover_position is not None:
+            return self.current_cover_position == 0
+
+        if self.current_cover_tilt_position is not None:
```

Comment:
Are there devices that only support tilt movement? We normally don't take tilt position into consideration for closed state.

Reply:
Yes, for example a pergola like the Brustor Pergola. In core, there is now also a demo device which only supports configuring `tilt_position`.

![image](https://user-images.githubusercontent.com/1424596/110605144-1b181d80-8189-11eb-8698-14533a37bfcf.png)

Reply:
Ok.